routes/post.js

```diff
@@ -0,0 +1,123 @@
+const http = require("http");
+const express = require("express")
+const cookieParser = require("cookie-parser")
+const bodyParser = require("body-parser")
+const ObjectId = require('mongoose').Types.ObjectId;
+const { Schema,
+    userSchema,
+    User,
+    connect,
+    sessionSchema,
+    Session,
+    postSchema,
+    Post,
+    replySchema,
+    Reply } = require("../database.js");
+
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+    try {
+        if (!req.session) {
+            res.clearCookie('session');
+            res.redirect('/');
+            return;
+        }
+        const currentUser = await User.findById(req.session.user);
+        res.render("newPost", currentUser)
+    } catch (err) {
+        res.render('error', { "error-message": err })
+    }
+})
+
+router.get("/:id", async (req, res) => {
+    try {
+        if(!ObjectId.isValid(req.params.id)){
+            res.redirect('/');
+            return;
+        }
+        const currentPost = await Post.findById(req.params.id);
+        if (!currentPost) {
+            res.redirect('/');
+            return;
+        }
+        const maker = await User.findById(currentPost.user.id);
+        const repliesOfThisPostQ = await Reply.find({ post: req.params.id })
+        const repliesOfThisPost = repliesOfThisPostQ.map(item => {
+            return {
+                reply: item.reply,
+                user: { name: item.user.name, id: item.user.id },
+                time: item.created
+            }
+        })
+        if (!req.session) {
+            res.render("post", {
+                post: currentPost,
+                user: maker,
+                replies: repliesOfThisPost
+            })
+            return;
+        }
+        const currentUser = await User.findById(req.session.user);
+        res.render("post", {
+            post: currentPost,
+            user: maker,
+            replies: repliesOfThisPost,
+            authenticated: true,
+            admin: currentUser.admin
+        })
```

Comment:
Last one on this topic, if you had `req.user` and you didn't have to `User.findById`, you could have collapsed the `if` branches completely with something like


```suggestion
        res.render("post", {
            post: currentPost,
            user: maker,
            replies: repliesOfThisPost,
            authenticated: !!req.user,
            admin: !!req.user && req.user.admin, // or the nicer shorthand: req.user?.admin ?? false
        })
```